Library/Homebrew/diagnostic.rb

```diff
@@ -785,6 +785,19 @@ def check_coretap_git_origin
               git -C "#{coretap_path}" remote set-url origin #{Formatter.url("https://github.com/Homebrew/homebrew-core.git")}
           EOS
         end
+
+        return if ENV["CI"] || ENV["JENKINS_HOME"]
+
+        branch = coretap_path.git_branch
+        return if branch.nil? || branch =~ /master/
```

Comment:
Should probably be `return if branch != "master"`  as `foo-master-bar` is an invalid branch name.